extensions/amp-analytics/0.1/analytics-root.js

```diff
@@ -279,6 +280,30 @@ export class AnalyticsRoot {
     });
   }
 
+  /**
+   * @param {string} selector DOM query selector.
+   * @return {!Promise<!Array<!Element>>} Element corresponding to the selector.
+   */
+  getElementsByScopedQuerySelectorAll_(selector) {
+    // Wait for document-ready to avoid false missed searches
+    return this.ampdoc.whenReady().then(() => {
+      const results = [];
+      const foundElements = scopedQuerySelectorAll(
+        dev().assertElement(this.ampdoc.getBody()),
```

Comment:
Could you confirm that if embed would get its own ampdoc? If not I think we still need to pass in the context

Reply:
Current implantation uses `this.getRoot().querySelector(selector)`, which I think is a better approach.